### PR TITLE
fix product availability date

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

**PR intended to be tested with API branch:** 

### Screenshots

Screenshot 2020-09-07 at 14 48 44

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

File: src/components/VisibilityCard/VisibilityCard.tsx [hunk 1]
```diff
@@ -239,11 +239,12 @@ export const VisibilityCard: React.FC<VisibilityCardProps> = props => {
...
                 const { value } = e.target;
-                if (!value) {
-                  onChange({
-                    target: { name: "availableForPurchase", value: null }
-                  });
-                }
+                onChange({
+                  target: {
+                    name: "availableForPurchase",
+                    value: value ? localizeDate(new Date().toString()) : null
```
Comment (maarcingebala) [inline src/components/VisibilityCard/VisibilityCard.tsx:245]: `productSetAvailabilityForPurchase` mutation in the response returns the product with `availableForPurchase` set by the backend. The current logic assumes that we set here the current date if `null` was passed as `startDate` in the mutation. If we change this logic in the backend, `new Date()` here will be invalid, won't it? Can we use the date returned by API here?

Comment (AlicjaSzu) [reply src/components/VisibilityCard/VisibilityCard.tsx:245]: yes we can, I wanted to avoid refetching all product data only for such small thing

**Review by orzechdev — APPROVED** [review]

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
If you go to the `src/products/mutations.ts` and modify `productSetAvailabilityForPurchase` to return 
```
product {
        id
        isAvailableForPurchase
      }
```
there wouldn't be any need for refetching whole product.

File: src/products/views/ProductUpdate/ProductUpdate.tsx [hunk 2]
```diff
@@ -177,13 +178,19 @@ export const ProductUpdate: React.FC<ProductUpdateProps> = ({ id, params }) => {
...
+          availableForPurchase: updatedProduct.availableForPurchase
+        }));
         notify({
           status: "success",
           text: intl.formatMessage({
             defaultMessage: "Product availability updated",
             description: "snackbar text"
           })
         });
+        refetch();
```
Comment (dominik-zeglen) [inline src/products/views/ProductUpdate/ProductUpdate.tsx:193]: It doesn't update by itself?

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
Mutation to update availability should not be fired if previous failed.

File: src/products/views/ProductUpdate/ProductUpdate.tsx (hunk 2 above)
Comment (AlicjaSzu) [reply src/products/views/ProductUpdate/ProductUpdate.tsx:193]: I forgot to remove it.

**Review by krzysztofwolski — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

**Pull request merged**